pkg/kubelet/kuberuntime/helpers_test.go

```diff
@@ -31,6 +31,56 @@ import (
 	utilpointer "k8s.io/utils/pointer"
 )
 
+func TestIsInitContainerFailed(t *testing.T) {
+	tests := []struct {
+		status      *kubecontainer.Status
+		isFailed    bool
+		description string
+	}{
+		{
+			status: &kubecontainer.Status{
+				State:    kubecontainer.ContainerStateExited,
+				ExitCode: 1,
+			},
+			isFailed:    true,
+			description: "Init container which state is exited and exitcode is non-zero shoud return true",
+		},
+		{
+			status: &kubecontainer.Status{
+				State: kubecontainer.ContainerStateUnknown,
+			},
+			isFailed:    true,
+			description: "Init container which state is unknown should return true",
+		},
+		{
+			status: &kubecontainer.Status{
+				State:    kubecontainer.ContainerStateExited,
+				ExitCode: 0,
+			},
+			isFailed:    false,
+			description: "Init container which state is exited and exitcode is zero should return false",
+		},
+		{
+			status: &kubecontainer.Status{
+				State: kubecontainer.ContainerStateRunning,
+			},
+			isFailed:    false,
+			description: "Init container which state is running should return false",
+		},
+		{
+			status: &kubecontainer.Status{
+				State: kubecontainer.ContainerStateCreated,
+			},
+			isFailed:    false,
+			description: "Init container which state is created should return false",
+		},
+	}
+	for i, test := range tests {
+		isFailed := isInitContainerFailed(test.status)
+		assert.Equal(t, test.isFailed, isFailed, "TestCase[%d]: %s", i, test.description)
```

Comment:
using 
```
t.Run(test.Description, func (testing.T) {
....
})
```

rather than putting the description in the assert would be cleaner here. (I'd also generally recommend using `require`  over `assert`, but I'm not too picky about it).

Reply:
Thanks very much. Do I need to modify it?